pkg/controller/gcppubsub/pubsub_wrapper.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcppubsub
```

Comment:
I would pull all of this out of the `controller` package and into a `knative/pkg/pubsub` package.

Reply:
Is this suggesting moving it to the knative/pkg repo? If so, can I do that as a follow up to this PR?

Reply:
why would we move it to knative/pkg?

Reply:
I am just suggesting moving it to `knative/eventing-sources/pkg/pubsub` because it is not code related to the controller.

Reply:
I would like to keep it here because it is an abstraction that will be used exclusively by this controller. I don't expect anything else in this repo to use the GCP PubSub client, so I don't think it is useful outside this package.

If anything else in the repo uses the GCP PubSub client, then we should move it (and its fake for unit testing) into `knative/eventing-sources/pkg`.